open_seq2seq/parts/convs2s/attention_wn_layer.py

```diff
@@ -0,0 +1,77 @@
+"""Implementation of the attention layer for convs2s. Inspired from https://github.com/tobyyouup/conv_seq2seq"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+import tensorflow as tf
+import math
+from open_seq2seq.parts.convs2s.ffn_wn_layer import FeedFowardNetworkNormalized
+
+
+class AttentionLayerNormalized(tf.layers.Layer):
+  """Attention layer for convs2s with weight normalization"""
+
+  def __init__(self, in_dim, embed_size, layer_id, add_res):
+    """initializes the attention layer.
+    It uses weight normalization for linear projections (Salimans & Kingma, 2016)  w = g * v/2-norm(v)
+
+    Args:
+      in_dim: int last dimension of the inputs
+      embed_size: int target embedding size
+      layer_id: int the id of current convolution layer
+      add_res: bool whether residual connection should be added or not
+    """
+    super(AttentionLayerNormalized, self).__init__()
+
+    self.add_res = add_res
+    with tf.variable_scope("attention_layer_" + str(layer_id)):
+
+      # linear projection layer to project the attention input to target space
+      self.tgt_embed_proj = FeedFowardNetworkNormalized(in_dim, embed_size, dropout=1.0,
+                                                          var_scope_name="att_linear_mapping_tgt_embed")
+
+      # linear projection layer to project back to the input space
+      self.out_proj = FeedFowardNetworkNormalized(embed_size, in_dim, dropout=1.0,
+                                                    var_scope_name="att_linear_mapping_out")
+
+  def call(self, input, target_embed, encoder_output_a, encoder_output_b, input_attention_bias):
+    """Calculates the attention vectors.
+
+    Args:
+      input: A float32 tensor with shape [batch_size, length, in_dim]
+      target_embed: A float32 tensor with shape [batch_size, length, in_dim] containing the target embeddings
+      encoder_output_a:
+      encoder_output_b:
+      input_attention_bias:
+    Returns:
```

Comment:
Need to have explicit blank line before Returns to render it correctly in the docs